oss-internship-2020/curl/examples/example1.cc

```diff
@@ -0,0 +1,78 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <libgen.h>
+#include <syscall.h>
+#include <iostream>
+#include "curl_sapi.sapi.h"
+#include "sandboxed_api/util/flag.h"
+
+/*
+int main() {
+    CURL *curl = curl_easy_init();
+    curl_easy_setopt(curl, CURLOPT_VERBOSE, 1l);
+    curl_easy_setopt(curl, CURLOPT_URL, "http://example.com");
+    curl_easy_perform(curl);
+    curl_easy_cleanup(curl);
+}
+*/
+
+class CurlApiSandboxEx1 : public CurlSandbox {
+
+  std::unique_ptr<sandbox2::Policy> ModifyPolicy(
+      sandbox2::PolicyBuilder*) override {
+
+    return sandbox2::PolicyBuilder()
+        .DangerDefaultAllowAll()
+        .AllowUnrestrictedNetworking()
+        .AddDirectory("/lib")
+        .BuildOrDie();
+  }
+
+};
+
+int main(int argc, char* argv[]) {
+
+  CurlApiSandboxEx1 sandbox;
+  
+  auto status_sandbox_init = sandbox.Init();
+  assert(status_sandbox_init.ok());
```

Comment:
Be explicit: Not everyone who is trying to use this will know that SAPI ships with its own `sapi::Status` class:

```c++
sapi::Status status = sandbox.Init();
```

It's also perfectly ok to reuse a single `status` variable.